lego/apps/users/tests/test_users_api.py

```diff
@@ -476,3 +477,95 @@ def test_own_penalties_serializer(self, mock_now):
+        self.test_user_url = (
+            f"/api/v1/users/{self.test_user.username}/update_photo_consent/"
+        )
+        PhotoConsent.objects.create(
+            user=self.test_user,
+            year=self.current_year,
+            semester=self.current_semester,
+            domain=WEBSITE_DOMAIN,
+            is_consenting=None,
+        )
+
+    def test_update_own_existing_consent(self):
+        self.client.force_authenticate(user=self.test_user)
+        response = self.client.post(
+            self.test_user_url,
+            {
+                "user": self.test_user.id,
+                "year": self.current_year,
+                "semester": self.current_semester,
+                "domain": WEBSITE_DOMAIN,
+                "isConsenting": True,
+            },
+        )
+        updated_consent = self.test_user.photo_consents.get(
+            year=self.current_year,
+            semester=self.current_semester,
+            domain=WEBSITE_DOMAIN,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(updated_consent.is_consenting, True)
+        self.assertIsNotNone(updated_consent.updated_at)
+
+    def test_update_own_non_existing_consent(self):
+        self.client.force_authenticate(user=self.test_user)
+        consent_exists = self.test_user.photo_consents.filter(
+            year=self.current_year,
+            semester=self.current_semester,
+            domain=SOCIAL_MEDIA_DOMAIN,
+        ).exists()
+        self.assertFalse(consent_exists)
+
+        response = self.client.post(
+            self.test_user_url,
+            {
+                "user": self.test_user.id,
+                "year": self.current_year,
+                "semester": self.current_semester,
+                "domain": SOCIAL_MEDIA_DOMAIN,
+                "isConsenting": False,
+            },
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        created_consent = self.test_user.photo_consents.get(
+            year=self.current_year,
+            semester=self.current_semester,
+            domain=SOCIAL_MEDIA_DOMAIN,
+        )
+        self.assertEqual(created_consent.is_consenting, False)
+
+    def test_update_other_user_consent(self):
+        self.client.force_authenticate(user=self.other_user)
+
+        response = self.client.post(
+            self.test_user_url,
+            {
+                "user": self.test_user.id,
+                "year": self.current_year,
+                "semester": self.current_semester,
+                "domain": WEBSITE_DOMAIN,
+                "isConsenting": False,
+            },
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        consent = self.test_user.photo_consents.get(
+            year=self.current_year,
+            semester=self.current_semester,
+            domain=WEBSITE_DOMAIN,
+        )
+        self.assertEqual(consent.is_consenting, None)
```

Comment:
Could use assertIsNone(), but that is nitpicking.

Reply:
Done ✅